pkg/controller/perconaservermongodb/getters.go

```diff
@@ -105,6 +105,27 @@ func (r *ReconcilePerconaServerMongoDB) getMongodStatefulsets(cr *api.PerconaSer
 	return list, err
 }
 
+func (r *ReconcilePerconaServerMongoDB) getNonMongosStatefulsets(cr *api.PerconaServerMongoDB) (appsv1.StatefulSetList, error) {
+	list := appsv1.StatefulSetList{}
+	filteredList := appsv1.StatefulSetList{}
+
+	err := r.client.List(context.TODO(),
+		&list,
+		&client.ListOptions{
+			Namespace:     cr.Namespace,
+			LabelSelector: labels.SelectorFromSet(clusterLabels(cr)),
+		},
+	)
+
+	for _, sts := range list.Items {
+		if sts.Labels["app.kubernetes.io/component"] != "mongos" {
+			filteredList.Items = append(filteredList.Items, sts)
+		}
+	}
+
+	return filteredList, err
```

Comment:
Rather than filtering the list manually, we can use selectors for excluding too. Please see my commit for the implementation: https://github.com/percona/percona-server-mongodb-operator/commit/980ea036a9919443b605b4995a92e84956177b7b

Also, I think `getStatefulsetsExceptMongos` is more readable than `getNonMongosStatefulsets`